src/hotspot/share/gc/shenandoah/shenandoahConcurrentGC.cpp

```diff
@@ -1215,6 +1215,12 @@ void ShenandoahConcurrentGC::op_final_updaterefs() {
     heap->verifier()->verify_roots_in_to_space();
   }
 
+  // This needs to see is_aging_cycle
```

Comment:
This comment is a bit unclear.  Are we saying we need the heap state to capture whether this is an aging cycle, because that will influence which regions get aged at the end of update refs?

Reply:
No, that was more of a 'note to self' to make sure the method calls didn't get re-ordered as part of this refactoring. I'll update the comment.